internal/scheduler/gang_scheduler.go

```diff
@@ -216,20 +214,9 @@ func (sch *GangScheduler) tryScheduleGang(ctx *armadacontext.Context, gctx *sche
 	return
 }
 
-func clearNodeBindings(jctx *schedulercontext.JobSchedulingContext) {
-	if jctx.PodSchedulingContext != nil {
-		// Clear any node bindings on failure to schedule.
-		jctx.PodSchedulingContext.NodeId = ""
-	}
-}
-
 func (sch *GangScheduler) tryScheduleGangWithTxn(_ *armadacontext.Context, txn *memdb.Txn, gctx *schedulercontext.GangSchedulingContext) (ok bool, unschedulableReason string, err error) {
 	if ok, err = sch.nodeDb.ScheduleManyWithTxn(txn, gctx.JobSchedulingContexts); err == nil {
 		if !ok {
-			for _, jctx := range gctx.JobSchedulingContexts {
-				clearNodeBindings(jctx)
```

Comment:
This now happens further up in the call tree, when we set `UnschedulableReason` for these jobs; I made this change because I prefer mutating both fields in tandem.